### PR TITLE
Add more logging to GetAccounts by id

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #

**Changes proposed in this request**

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 1]
```diff
@@ -171,6 +171,15 @@ private async Task<IEnumerable<IAccount>> GetAccountsInternalAsync(ApiIds apiId,
         {
             Guid correlationId = Guid.NewGuid();
             RequestContext requestContext = CreateRequestContext(correlationId, cancellationToken);
+            
+            if (requestContext.Logger.IsLoggingEnabled(LogLevel.Info))
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/ClientApplicationBase.cs:175]: So we usually check this before creating a logger message that has quite a bit of overhead. This doesn't seem to be too taxing so cant we let the logger adapter determine if this should be logged or not?

File: tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/ClientCredentialsTests.WithRegion.cs [hunk 2]
```diff
@@ -125,17 +125,18 @@ private IConfidentialClientApplication BuildCCA(
...
             var builder = ConfidentialClientApplicationBuilder.Create(settings.ClientId);
-            if (useClaims)
-            {
-                builder.WithClientAssertion(() => GetSignedClientAssertionUsingMsalInternal(settings.ClientId, GetClaims(settings)));
-            }
-            else
-            {
-                builder.WithCertificate(settings.GetCertificate());
-            }
+            //if (useClaims)
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/ClientCredentialsTests.WithRegion.cs:128]: Dead code?

**Review by trwalke — APPROVED** [review]
Minor comments

File: src/client/Microsoft.Identity.Client/ClientApplicationBase.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/ClientApplicationBase.cs:175]: This is in the hot path ... Any optimization counds. Besides at line 179-180 all those string interpolations are not free and will lead to some allocations etc.

**Review by gladjohn — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/ClientCredentialsTests.WithRegion.cs (hunk 2 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/ClientCredentialsTests.WithRegion.cs:128]: Good catch, thx.

**Pull request merged**